wallet/src/de/schildbach/wallet/ui/dashpay/TransactionsViewModel.kt

```diff
@@ -0,0 +1,95 @@
+package de.schildbach.wallet.ui.dashpay
+
+import android.app.Application
+import androidx.lifecycle.AndroidViewModel
+import androidx.lifecycle.MediatorLiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.viewModelScope
+import de.schildbach.wallet.Constants
+import de.schildbach.wallet.WalletApplication
+import de.schildbach.wallet.data.DashPayProfile
+import de.schildbach.wallet.data.UsernameSortOrderBy
+import kotlinx.coroutines.launch
+import org.bitcoinj.core.Context
+import org.bitcoinj.core.Sha256Hash
+import org.bitcoinj.core.Transaction
+import org.bitcoinj.core.TransactionConfidence
+import org.bitcoinj.wallet.Wallet
+import java.util.*
+import kotlin.collections.HashMap
+
+class TransactionsViewModel(application: Application) : AndroidViewModel(application) {
+
+    enum class Direction {
+        RECEIVED, SENT
+    }
+
+    val direction = MutableLiveData<Direction?>()
+    val transactionsLiveData = MediatorLiveData<Pair<List<Transaction>,
+            Map<Sha256Hash, DashPayProfile>>>()
+    private val balanceLiveData = WalletBalanceLiveData()
+
+    init {
+        load()
+        transactionsLiveData.addSource(direction) {
+            load()
+        }
+        transactionsLiveData.addSource(balanceLiveData) {
+            load()
+        }
+    }
+
+    fun load(wallet: Wallet = WalletApplication.getInstance().wallet) {
+        viewModelScope.launch {
+            Context.propagate(Constants.CONTEXT)
+
+            val contactsTransactions: HashMap<Sha256Hash, DashPayProfile> = hashMapOf()
+            val contactsByIdentity: HashMap<String, DashPayProfile> = hashMapOf()
+            val platformRepo = PlatformRepo.getInstance()
+            val userIdentity = platformRepo.getBlockchainIdentity()
+            if (userIdentity != null) {
+                val contacts = PlatformRepo.getInstance().searchContacts("",
+                        UsernameSortOrderBy.LAST_ACTIVITY, false)
+                contacts.data?.forEach {
+                    contactsByIdentity[it.dashPayProfile.userId] = it.dashPayProfile
+                }
+            }
+
+            val transactions = wallet.getTransactions(true)
+            val filteredTransactions = arrayListOf<Transaction>()
+            transactions.filterTo(filteredTransactions, {
+                val sent = it.getValue(wallet).signum() < 0
+                val isInternal = it.purpose == Transaction.Purpose.KEY_ROTATION
+                direction.value == Direction.RECEIVED && !sent
+                        && !isInternal || direction.value == null ||
+                        direction.value == Direction.SENT && sent && !isInternal
+            })
```

Comment:
This part will be changed by https://github.com/dashevo/dash-wallet/pull/526 but is ok for this PR